lib/controllers/animals.js

```diff
@@ -0,0 +1,52 @@
+const { Router } = require('express');
+const Animal = require('../models/Animal');
+module.exports = Router()
+
+  .get('/', async (req, res, next) => {
+    try {
+      const animalsData = await Animal.getAll();
+      res.json(animalsData);
+    } 
+    catch (error) {
+      next (error);
+    }
+  })
+
+  .get('/:id', async (req, res, next) => {
```

Comment:
on your detail route, its good to add in a check to make sure you actually have an item with the matching id. For example, if we were doing sodas:
```js
    try {
      const data = await Soda.getById(req.params.id);
      if (!data) {
        next();
      }
      res.json(data);
    } catch (e) {
      next(e);
    }
```
Notice that if I don't get any data back from the database, I call the next middleware function which will 404